jigglypuffRL/deeprl/ppo1/ppo1.py

```diff
@@ -0,0 +1,229 @@
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim as opt
+from torch.autograd import Variable
+from torch.distributions import Categorical
+import gym
+
+from jigglypuffRL.common import (
+    MlpPolicy,
+    MlpValue,
+    get_policy_from_name,
+    get_value_from_name,
+)
+
+
+class PPO1:
+    """
+    Proximal Policy Optimization algorithm (Clipped policy).
+    Paper: https://arxiv.org/abs/1707.06347
+    :param policy: (str) The policy model to use (MlpPolicy)
+    :param value: (str) The value function model to use (MlpValue)
+    :param env: (Gym environment) The environment to learn from
+    :param timesteps_per_actorbatch: (int) timesteps per actor per update
+    :param gamma: (float) discount factor
+    :param clip_param: (float) clipping parameter epsilon
+    :param actor_batchsize: (int) trajectories per optimizer epoch
+    :param epochs: (int) the optimizer's number of epochs
+    :param lr_policy: (float) policy network learning rate
+    :param lr_value: (float) value network learning rate
+    :param policy_copy_interval: (int) number of optimizer before copying params from new policy to old policy
+    :param tensorboard_log: (str) the log location for tensorboard (if None, no logging)
+    :param seed (int): seed for torch and gym
+    :param device (str): device to use for tensor operations; 'cpu' for cpu and 'cuda' for gpu
+    """
+
+    def __init__(
+        self,
+        policy,
+        value,
+        env,
+        timesteps_per_actorbatch=200,
+        gamma=0.99,
+        clip_param=0.2,
+        actor_batch_size=8,
+        epochs=1000,
+        lr_policy=0.001,
+        lr_value=0.005,
+        policy_copy_interval=20,
+        tensorboard_log=None,
+        seed=0,
+        device="cuda",
+    ):
+        self.policy = policy
+        self.value = value
+        self.env = env
+        self.timesteps_per_actorbatch = timesteps_per_actorbatch
+        self.gamma = gamma
+        self.clip_param = clip_param
+        self.actor_batch_size = actor_batch_size
+        self.epochs = epochs
+        self.lr_policy = lr_policy
+        self.lr_value = lr_value
+        self.tensorboard_log = tensorboard_log
+        self.seed = seed
+        self.policy_copy_interval = policy_copy_interval
+
+        # Assign device
+        if "cuda" in device and torch.cuda.is_available():
+            self.device = torch.device(device)
+        else:
+            self.device = torch.device("cpu")
+
+        # Assign seed
+        self.env.seed(seed)
+        torch.manual_seed(seed)
```

Comment:
There are some seed settings missing here right?
```np.random.seed(), random.seed(), torch.backends.cudnn.deterministic = True, torch.backends.cudnn.benchmark = False```
Check https://pytorch.org/docs/stable/notes/randomness.html
Maybe something like this would make more sense:
```if seed!=None:
    torch.manual_seeds()
    np.random....
    .
    .
    .```